src/contrib/cluster/Akka.Cluster.Sharding/ShardRegion.cs

```diff
@@ -607,18 +606,23 @@ protected override bool Receive(object message)
                 case IShardRegionQuery srq:
                     HandleShardRegionQuery(srq);
                     return true;
-                case RestartShard _:
-                    DeliverMessage(message, Sender);
+                case RestartShard restart:
+                    DeliverRestartShard(restart, Sender);
```

Comment:
Added a dedicated method for restarting shards so we could simplify the `DeliverMessage` handling.